### PR TITLE
Limit Memory Impact and Upgrade dependencies

### PR DESCRIPTION
**Pull request by fredericBregier**
Limit memory impact
    
- For Server, change Web, Web Admin, REST V1 and V2 configurations (threads)
    
   - For all, the old (unused) argument `memorylimit` to a value like 100000000 (100MB)
     can be used to limit Memory impact
   - For REST V2: improve configuration, in particular network fine configuration and workers
   - One could add the standard option `-Xmx1g` to limit to 1GB of memory.
     It is possible for not heavy loaded Waarp server to limit to 512 MB (`-Xmx512m`), but
     one should not make it smaller.
  - Adapt the XML configuration according to context:
    - `runlimit` to a number between 100 to 1000 (at most 1000)
    - `serverthread` to 1 to 32 (at most 2x number of cores)
    - `clientthread` to 10 to 100 (at most 10x serverthread)
    - `memorylimit` to a number between 100000 to 1000000 (100MB to 1GB) for web and Rest purpose only
    - Don't activate unused sub-services such as: REST (V1 and V2), SNMP, THRIFT, CPU limitation, Bandwidth limitation
    
- For client only, one can specify in XML configuration file the following items to limit
  memory impact
    
   - `runlimit` to 1
   - `serverthread` to 1
   - `clientthread` to 1
   - One could add the standard option `-Xmx128m` (128MB but not less) for such limited client
     with no extra support activated (no CPU limitation, others like Thrift, Snmp and Rest
     support being off for client). Note however that it is still recommended to allow at
     least 512 MB of memory (`-Xmx512m`).

Doing the following works for a Server according to related needs:

- Create IT test for Big file
- Create IT test for REST/Web intensive
    
In reference to issue #52 (OOME on Pull request).

Fix dependencies and minor for tests

- Upgrade dependencies
- Fix minor cosmetic for tests
- Reduce memory and cpu usage in Hash computations

### CODE REVIEW

**Comment by fredericBregier** [comment]
Note: waiting for Netty upgrade in order to include the corrections from Netty

**Comment by fredericBregier** [comment]
@bcarlin Ready

File: doc/waarp-r66/source/changes.rst [hunk 1]
```diff
@@ -50,12 +50,17 @@ Correctifs
...
+  Diminution de l'empreinte mémoire pour le cas des clients simples et diminution
+  de la mémoire côté serveur pour les parties Web et REST.
+  (issue [`#52 <https://github.com/waarp/Waarp-All/issues/52>`__])
 
 .. code-block:: xml
 
   <statusListener
     class="org.waarp.common.logging.PrintOnlyWarningLogbackStatusListener" />
 
+- Mise à jour des dépendances
```
Comment (bcarlin) [inline doc/waarp-r66/source/changes.rst]: I propose to add separate entries in the changelog for the changes of default values introduced in #51 (IMO, this is important enough to be highlighted).

```suggestion

.. code-block:: xml

  <statusListener
    class="org.waarp.common.logging.PrintOnlyWarningLogbackStatusListener" />

- [`#51 <https://github.com/waarp/Waarp-All/pull/51>`__]
  Diminution de l'empreinte mémoire pour le cas des clients simples et diminution
  de la mémoire côté serveur pour les parties Web et REST.
  (issue [`#52 <https://github.com/waarp/Waarp-All/issues/52>`__])
- [`#51 <https://github.com/waarp/Waarp-All/pull/51>`__] Si aucun argument
  ``-Xms`` n'est passé à la JVM lors du démarrage, une valeur par défaut de
  1024Mo est appliquée.
- [`#51 <https://github.com/waarp/Waarp-All/pull/51>`__] Les valeurs par défaut
  des limitations de bande passante ont changées : La limitation globale par
  défaut est maintenant de 100Gbps, et celle par connexion est de 1Gbps (ces
  valeurs peuvent être ajustées dans les fichiers de configuration).
- [`#51 <https://github.com/waarp/Waarp-All/pull/51>`__] La valeur par défaut
  de la RAM maximale utilisée par les services WEB et REST a été abaissée à 1Go
  (au lieu de 4Go) (cette valeur peut être ajustée dans les fichiers de
  configuration).
- Mise à jour des dépendances
```

**Review by bcarlin — APPROVED** [review]

File: doc/waarp-r66/source/changes.rst (hunk 1 above)
Comment (fredericBregier) [reply doc/waarp-r66/source/changes.rst]: Done

**Pull request merged**